EPMEDU-738 [Android] UI: Switcher between Dogs and Cats

VEZE opened the pull request:
- Added Switch component between Dogs, Cats
- Added switch animation, styled ripple
- Added darkMode support

| Light mode  | Dark mode  |
| ------------- | ------------- |
| <video src="https://user-images.githubusercontent.com/11144164/177144938-2f348652-8780-4df1-b3bc-034e85d3aa50.mp4">  | <video src="https://user-images.githubusercontent.com/11144164/177144930-e0deb871-94fc-433d-b052-3d6960805176.mp4">|

MaksimNasalevich committed:
EPMEDU-728
- created sign in screen
- added animeal button

MaksimNasalevich committed:
EPMEDU-728
- fixed detekt issues

MaksimNasalevich committed:
EPMEDU-728
- Added splash screen

MaksimNasalevich committed:
Merge branch 'develop' of https://github.com/AnimealProject/animeal_android into feature/EPMEDU-728_Sign_up_page

 Conflicts:
	app/src/main/java/com/epmedu/animeal/tabs/TabActivity.kt
	settings.gradle.kts

Commit:
input field init

Commit:
font family added

Commit:
reusable name/surname input

Commit:
reusable name/surname input

Commit:
reusable formatted input

Commit:
Merge branch 'develop' of https://github.com/AnimealProject/animeal_android into feature/EPMEDU-748_Input_field_name_surname

Commit:
detect fixes

Commit:
Detekt fixes

egorikftp committed:
EPMEDU-755: [Android] [UI] Align Tab Bar with designs

egorikftp committed:
Disable NoWildcardImports rule

VEZE committed:
EPMEDU-738 [Android] UI: Switcher between Dogs and Cats [Added & styled switch between Cats, Dogs.]

VEZE committed:
EPMEDU-738 [Android] UI: Switcher between Dogs and Cats [Added & styled switch between Cats, Dogs.]

VEZE committed:
EPMEDU-738 [Android] UI: Switcher between Dogs and Cats [Fixed naming of indicator method]

VEZE committed:
EPMEDU-738 [Android] UI: Switcher between Dogs and Cats [Fixed naming of indicator method]

VEZE committed:
EPMEDU-738 [Android] UI: Switcher between Dogs and Cats [Fixed naming of Tab method]

VEZE committed:
EPMEDU-738 [Android] UI: Switcher between Dogs and Cats [Fixed naming of Tab method]

VEZE committed:
EPMEDU-738 [Android] UI: Switcher between Dogs and Cats [Fixed naming of Switch]

VEZE committed:
EPMEDU-738 [Android] UI: Switcher between Dogs and Cats [Fixed naming of Switch]

Commit:
Merge branch 'develop' of https://github.com/AnimealProject/animeal_android into feature/EPMEDU-748_Input_field_name_surname

 Conflicts:
	base/src/main/java/com/epmedu/animeal/base/theme/Color.kt

VEZE committed:
EPMEDU-738 [Android] UI: Switcher between Dogs and Cats [Added named arguments, changed "hardcoded" colors to MatherialTheme, fixed ripple shape]

VEZE committed:
EPMEDU-738 [Android] UI: Switcher between Dogs and Cats [Added named arguments, changed "hardcoded" colors to MatherialTheme, fixed ripple shape]

VEZE committed:
EPMEDU-738 [Android] UI: Switcher between Dogs and Cats [Removed component privacy]

VEZE committed:
EPMEDU-738 [Android] UI: Switcher between Dogs and Cats [Removed component privacy]

VEZE committed:
EPMEDU-738 [Android] UI: Switcher between Dogs and Cats [Removed divider with shadow]

VEZE committed:
EPMEDU-738 [Android] UI: Switcher between Dogs and Cats [Removed divider with shadow]

VEZE committed:
EPMEDU-738 [Android] UI: Switcher between Dogs and Cats [Added example to HomeScreen]

VEZE committed:
EPMEDU-738 [Android] UI: Switcher between Dogs and Cats [fixed detekt]

VEZE committed:
EPMEDU-738 [Android] UI: Switcher between Dogs and Cats [fixed detekt]

egorikftp committed:
Rename colors

MaksimNasalevich committed:
EPMEDU-728
- Added extensions
- Added navigation module
- moved all resources into resources module

MaksimNasalevich committed:
EPMEDU-728
- Added extensions

MaksimNasalevich committed:
Merge branch 'develop' of https://github.com/AnimealProject/animeal_android into feature/EPMEDU-728_Sign_up_page

 Conflicts:
	app/build.gradle.kts
	app/src/main/java/com/epmedu/animeal/tabs/TabActivity.kt
	base/src/main/java/com/epmedu/animeal/base/theme/Color.kt
	base/src/main/java/com/epmedu/animeal/base/theme/Theme.kt
	feature-more/src/main/java/com/epmedu/animeal/feature/more/MoreScreen.kt
	feature-more/src/main/java/com/epmedu/animeal/feature/more/Screens.kt
	feature/more/src/main/res/values/strings.xml
	library/foundation/src/main/java/com/epmedu/animeal/foundation/topbar/AnimealShortButton.kt
	library/foundation/src/main/java/com/epmedu/animeal/foundation/topbar/TopBar.kt
	settings.gradle.kts

egorikftp committed:
Merge pull request #19 from AnimealProject/feature/EPMEDU-755_Android_UI_Align_Tab_Bar_with_designs

EPMEDU-755: [Android] [UI] Align Tab Bar with designs

MaksimNasalevich committed:
EPMEDU-728
- Merged conflicts

MaksimNasalevich committed:
EPMEDU-728
- fixed detekt

MaksimNasalevich committed:
Merge branch 'develop' of https://github.com/AnimealProject/animeal_android into feature/EPMEDU-728_Sign_up_page

 Conflicts:
	app/src/main/java/com/epmedu/animeal/tabs/TabActivity.kt
	base/src/main/java/com/epmedu/animeal/base/theme/Color.kt
	base/src/main/java/com/epmedu/animeal/base/theme/Theme.kt

MaksimNasalevich committed:
EPMEDU-728
- fixed conflicts

MaksimNasalevich committed:
EPMEDU-728
- fixed conflicts

MaksimNasalevich committed:
EPMEDU-728
- added dark colors

MaksimNasalevich committed:
EPMEDU-728
- updated github actions

MaksimNasalevich committed:
EPMEDU-728
- moved preview to the bottom

MaksimNasalevich committed:
EPMEDU-728
- fixed issues

MaksimNasalevich committed:
EPMEDU-728
- removed UnusedPrivateMember from previews

MaksimNasalevich committed:
EPMEDU-728
- renamed Spacers, moved shortButton to button package

FlinForm committed:
EPMEDU-754: [Android] Add Firebase Crashlytics to project

FlinForm committed:
Fix ci-workflow file

FlinForm committed:
EPMEDU-754: Fix ci-workflow file

Fileek commented on base/src/main/java/com/epmedu/animeal/base/ui/AnimealSwitch.kt:
```diff
@@ -0,0 +1,171 @@
+            .wrapContentSize(align = Alignment.BottomStart)
+            .offset(x = indicatorLeft)
+            .width(width = indicatorRight - indicatorLeft)
+            .fillMaxSize()
```
Could be replaced by fillMaxHeight

VEZE replied:
changed to FillMaxHeight

Fileek commented on base/src/main/java/com/epmedu/animeal/base/ui/AnimealSwitch.kt:
```diff
@@ -0,0 +1,171 @@
+        Modifier
+            .wrapContentSize(align = Alignment.BottomStart)
+            .offset(x = indicatorLeft)
+            .width(width = indicatorRight - indicatorLeft)
```
It's clear that you pass width, you don't need to use named argument here.

VEZE replied:
removed

Fileek commented on base/src/main/java/com/epmedu/animeal/base/ui/AnimealSwitch.kt:
```diff
@@ -0,0 +1,171 @@
+                color = MaterialTheme.colors.primary,
+                shape = RoundedCornerShape(9.dp)
+            )
+            .zIndex(zIndex = -1f)
```
It's clear that you pass zIndex, you don't need to use named argument here.

VEZE replied:
removed

Commit:
Comment fixes

Commit:
Comment fixes

MaksimNasalevich committed:
EPMEDU-728
- changed default height

Commit:
Comment fixes

VEZE committed:
EPMEDU-738 [Android] UI: Switcher between Dogs and Cats [fixed detekt]

VEZE committed:
EPMEDU-738 [Android] UI: Switcher between Dogs and Cats [fixed detekt]

Commit:
Comment fixes

Commit:
Merge branch 'develop' of https://github.com/AnimealProject/animeal_android into feature/EPMEDU-748_Input_field_name_surname

Commit:
WildCardImports fix

DoTheMonkeyBusiness committed:
EPMEDU-728: SignIn page

EPMEDU-728: SignIn page

- created SignIn screen
- created Splash screen
- added navigation module
- added extensions module
- added foundation module
- added common module
- added resources module
- moved all resources into resources module

FlinForm committed:
EPMEDU-754: Fix PR comments

FlinForm committed:
Merge pull request #24 from AnimealProject/feature/PMEDU-754-Add_Firebase_Crashlytics_to_project

EPMEDU-754: [Android] Add Firebase Crashlytics to project

Commit:
Merge branch 'develop' of https://github.com/AnimealProject/animeal_android into feature/EPMEDU-748_Input_field_name_surname

 Conflicts:
	app/src/main/java/com/epmedu/animeal/tabs/TabActivity.kt
	base/src/main/java/com/epmedu/animeal/base/theme/Color.kt
	feature/tabs/src/main/java/com/epmedu/animeal/tabs/presentation/TextInputField.kt

Commit:
Input field fixes

Commit:
Reverted view imports

Commit:
Reverted view imports

Commit:
Reverted view imports

egorikftp committed:
Optimize Button component

Commit:
Fixed preview private

egorikftp committed:
Change content color for sign in buttons

Commit:
Minor fixes

egorikftp committed:
Use color onPrimary for buttons

egorikftp committed:
Merge pull request #28 from AnimealProject/Optimize_button_component

Optimize button component

Commit:
Merge branch 'develop' of https://github.com/AnimealProject/animeal_android into feature/EPMEDU-748_Input_field_name_surname

 Conflicts:
	base/src/main/java/com/epmedu/animeal/base/theme/Color.kt

Muhammadali-mk committed:
Merge pull request #17 from AnimealProject/feature/EPMEDU-748_Input_field_name_surname

Feature/epmedu 748 input field name surname

egorikftp commented on base/src/main/java/com/epmedu/animeal/base/ui/AnimealSwitch.kt:
```diff
@@ -0,0 +1,171 @@
+ * @param onTabSelected Called when the tab is switched.
+ * @param modifier The [Modifier].
+ */
+@Composable
```
could you please move into foundation module?

egorikftp commented on base/src/main/res/values/strings.xml:
```diff
@@ -1,4 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources>
     <string name="back">Back</string>
+    <string name="tab_cats_title">Cats</string>
```
and this into resources module

VEZE replied:
moved to resources module

FlinForm committed:
EPMEDU-733: [Android] Get the permission to use native GPS

FlinForm committed:
Merge pull request #20 from AnimealProject/feature/EPMEDU-733-Get_the_permission_to_use_native_GPS

EPMEDU-733: [Android] Get the permission to use native GPS

FlinForm commented on base/src/main/java/com/epmedu/animeal/base/ui/AnimealSwitch.kt:
```diff
@@ -0,0 +1,171 @@
+
+    val transition = updateTransition(
+        targetState = tab,
+        label = "Tab indicator"
```
Can we move this to resources or constants, depending on whether this text is shown to user or no.

VEZE replied:
Not showed to user

egorikftp replied:
agree, it's not necessary

FlinForm replied:
I mean that if that string is not shown to user it has to be moved to top level constants, otherwise it needs to be stored in resource files

VEZE replied:
moved to constants.

FlinForm commented on base/src/main/java/com/epmedu/animeal/base/ui/AnimealSwitch.kt:
```diff
@@ -0,0 +1,171 @@
+    )
+
+    val indicatorLeft by transition.animateDp(
+        label = "Indicator left"
```
Can we move this to resources or constants, depending on whether this text is shown to user or no.

VEZE replied:
Not showed to user

FlinForm commented on base/src/main/java/com/epmedu/animeal/base/ui/AnimealSwitch.kt:
```diff
@@ -0,0 +1,171 @@
+    }
+
+    val indicatorRight by transition.animateDp(
+        label = "Indicator right"
```
Can we move this to resources or constants, depending on whether this text is shown to user or no.

VEZE replied:
Not showed to user

VEZE committed:
Resolved merge conflicts, moved to foundation package

VEZE committed:
Merge remote-tracking branch 'origin/feature/EPMEDU-735_Switch_between_Dogs_and_Cats' into feature/EPMEDU-735_Switch_between_Dogs_and_Cats

# Conflicts:
#	feature/tabs/src/main/java/com/epmedu/animeal/tabs/presentation/Screens.kt
#	library/foundation/src/main/java/com/epmedu/animeal/foundation/topbar/AnimealSwitch.kt

VEZE committed:
Label text moved to constants

VEZE committed:
Removed string res from base module

VEZE commented:
merge corrections
